sys/include/hashes/pbkdf2.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @defgroup    sys_hashes_pbkdf2 PBKDF2
+ * @ingroup     sys_hashes
+ * @brief       PBKDF2 key derivation implementation.
+ * @{
+ *
+ * @file
+ * @brief       PBKDF2 key derivation implementation.
+ *
+ * @author      Juan I Carrano <j.carrano@fu-berlin.de>
+ *
+ * @}
+ */
+
+#ifndef HASHES_PBKDF2_H
+#define HASHES_PBKDF2_H
+
+#include "hashes/sha256.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   PBKDF2 key size length
+ *
+ * @note Currently only one derived key length is supported (32)
+ */
+#define PBKDF2_KEY_SIZE SHA256_DIGEST_LENGTH
+
+/**
+ * @brief Create a key from a password and hash using PBKDF2.
+ *
+ * @param[in]   password        password pointer
+ * @param[in]   password_len    length of password
+ * @param[in]   salt            salt pointer
+ * @param[in]   salt_len        salt length
+ * @param[in]   iterations      number of rounds. Must be >1.
```

Comment:
Can we recommend some value here?

Reply:
Indeed although the recommended value is not staright forward as nicely described in https://security.stackexchange.com/questions/3959/recommended-of-iterations-when-using-pkbdf2-sha256 and https://cryptosense.com/blog/parameter-choice-for-pbkdf2/, maybe 10000 would be a good recommendation based on these.

Reply:
@benpicco I already added the recommendation here!

Reply:
Ah sorry, didn't see that